### PR TITLE
Responder shall terminate the encapsulated request flow when Requeste…

### PR DESCRIPTION
**Pull request by Zhiqiang520**
…r delivers encapsulated ERROR message.

Fix: #1791

### CODE REVIEW

File: library/spdm_responder_lib/libspdm_rsp_encap_response.c [hunk 1]
```diff
@@ -124,6 +124,15 @@ static libspdm_return_t libspdm_process_encapsulated_response(
     bool need_continue;
     libspdm_encap_response_struct_t encap_response_struct;
 
+    /* the Requester delivers an encapsulated ERROR message, the Responder shall terminates the encapsulated request flow. */
+    if (encap_response != NULL) {
+        if (((const spdm_message_header_t *)encap_response)->request_response_code == SPDM_ERROR) {
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_encap_response.c:129]: I dont think this line code is proper here.

It should be in a generic request_response_code checker/dispatcher.

**Review by jyao1 — CHANGES_REQUESTED** [review]
Please update libspdm_handle_encap_error_response_main().

File: unit_test/test_spdm_responder/encap_challenge.c [hunk 2]
```diff
@@ -100,6 +100,7 @@ void libspdm_test_responder_encap_challenge_case2(void **state)
     size_t response_size;
     void *data;
     size_t data_size;
+    bool need_continue;
```
Comment (jyao1) [inline unit_test/test_spdm_responder/encap_challenge.c:103]: why we need change test case?

need_continue should not be assigned if the status is ERROR

File: library/spdm_responder_lib/libspdm_rsp_encap_get_digests.c [hunk 3]
```diff
@@ -78,9 +78,10 @@ libspdm_return_t libspdm_process_encap_response_digest(
         status = libspdm_handle_encap_error_response_main(
             spdm_context,
             spdm_response->header.param1);
-        if (LIBSPDM_STATUS_IS_ERROR(status)) {
-            return status;
-        }
+        /* if the Requester delivers an encapsulated ERROR message, the Responder shall
+         * continue the current encapsulated communication and not move to the next request. */
+        *need_continue = true;
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_encap_get_digests.c:83]: need_continue should be assigned only if SUCCESS.

File: library/spdm_responder_lib/libspdm_rsp_encap_response.c (hunk 1 above)
Comment (Zhiqiang520) [reply library/spdm_responder_lib/libspdm_rsp_encap_response.c]: I have updated libspdm_handle_encap_error_response_main().

File: library/spdm_responder_lib/libspdm_rsp_encap_get_digests.c (hunk 3 above)
Comment (Zhiqiang520) [reply library/spdm_responder_lib/libspdm_rsp_encap_get_digests.c]: I have updated, please check.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**